src/components/footer.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { SelectOptions } from '../types/SelectOptions';
+
+type Props = {
+  filterTodos: (option: SelectOptions) => void,
+  todosCount: number,
+  completedTodosCount: number,
+};
+
+export const Footer: React.FC<Props> = ({
+  filterTodos,
+  todosCount,
+  completedTodosCount,
+}) => {
+  const [selectedFilter, setSelectedFilter] = useState(SelectOptions.ALL);
+
+  return (
+    <footer className="todoapp__footer">
+      <span className="todo-count">
+        {todosCount}
+        {' '}
+        items left
+      </span>
+
+      <nav className="filter">
+        <a
+          href="#/"
+          className={classNames(
+            'filter__link', {
+              selected: selectedFilter === SelectOptions.ALL,
+            },
+          )}
+          onClick={() => {
+            setSelectedFilter(SelectOptions.ALL);
+            filterTodos(SelectOptions.ALL);
+          }}
```

Comment:
Move it to a separate function

Reply:
> * Why here is `0` while todo is loading? Wait until todos will be loaded before show smth

Because It was short condition before render and it worked in that way. Now condition longer it dosent show 0